src/tests/class_tests/openms/source/DataValue_test.cpp

```diff
@@ -623,10 +624,15 @@ START_SECTION((void setUnit(const String& unit)))
 
   a1.setUnit("ppm");
   TEST_EQUAL(a1.getUnit(), "ppm")
+  auto tmpref = a1.getUnit();
 
   a1.setUnit("kg");
   TEST_EQUAL(a1.getUnit(), "kg")
 
+  a1.setUnit(a1.getUnit());
+  TEST_EQUAL(a1.hasUnit(), true)
+  TEST_EQUAL(a1.getUnit(), "kg")
+  // note: tmpref has now been deleted (access is illegal)
```

Comment:
this may be the downside of the PR: it is now possible to invalidate references to the unit due to manual memory management